src/test/java/StringManipulationTest.java

```diff
@@ -118,32 +218,69 @@ public void testRestoreString1()
         assertEquals(restoreString, "rat");
     }
 
+    // This will test the method if it will swap the two words in the string.
+    // With the space seperating the word staying in it's place, the test
+    // will take each character from each side of the space and swap then for the 
+    // character on the other side of the space and output a swapped string of two words.
     @Test
     public void testRestoreString2()
     {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("four five");
+        int [] array;
+        array=new int[]{5,6,7,8,4,0,1,2,3};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "five four");
 
     }
 
+    // This will test the method to move all spaces in between the characters into 
+    // one large space after the first character. 
     @Test
     public void testRestoreString3()
     {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("a b c d e f g");
+        int [] array;
+        array=new int[]{0,1,7,2,8,3,9,4,10,5,11,6,12};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "a      bcdefg");
 
     }
 
+    // This will test if the method will throw an IllegalArgumentException().
+    // The method should accept an int array of equal size to the string that is 
+    // being manipulated. In this test case, the string has 24 characters, and the
+    // int array that is passed in only has 6 elements. Therefore, this is an invalid
+    // size of int array, throwing an illegal argument exception.
     @Test
     public void testRestoreString4()
     {
-        fail("Not yet implemented");
-
+    	manipulatedstring.setString("illegalargumentexception");
+        int [] array;
+        array=new int[]{5,4,3,2,1,0};
+        
+        assertThrows(IllegalArgumentException.class, () -> {
+        	manipulatedstring.restoreString(array);
+        });
     }
 
+    // This will test if the method will throw an IndexOutOfBoundsException().
```

Comment:
I believe this and the other previous test case is what I accidentally cut off in both of my submissions.

Good to test the out of bounds exception on array.

Reply:
Alvin, interesting that your test cases somehow got cut off, hopefully you get those back in!

I'm curious, I know that the assignment calls for a test on for IllegalArgumentException() and IndexOutOfBoundsException(), do you think there are other possible exceptions that this method could throw? ArrayIndexOutOfBoundsException(), perhaps?